communication/mavlink_waypoint_handler.cpp

```diff
@@ -1497,28 +1487,24 @@ static waypoint_local_struct_t waypoint_handler_set_waypoint_from_frame(Mavlink_
             break;
 
         case MAV_FRAME_LOCAL_NED:
-            waypoint_coor.pos[X] = current_waypoint->x;
-            waypoint_coor.pos[Y] = current_waypoint->y;
-            waypoint_coor.pos[Z] = current_waypoint->z;
-            waypoint_coor.heading = maths_deg_to_rad(current_waypoint->param4);
-            waypoint_coor.origin = coord_conventions_local_to_global_position(waypoint_coor);
+            wpt.position[X] = current_waypoint->x;
+            wpt.position[Y] = current_waypoint->y;
+            wpt.position[Z] = current_waypoint->z;
             break;
 
         case MAV_FRAME_MISSION:
             // Problem here: rec is not defined here
             //mavlink_msg_mission_ack_send(MAVLINK_COMM_0,rec->msg.sysid,rec->msg.compid,MAV_CMD_ACK_ERR_NOT_SUPPORTED);
```

Comment:
This method does not return a mav_result_t, so we cannot directly notify an unsupported frame. However it is used in `continue_to_next_waypoint`, and `start_stop_navigation` which could notify the ground station. It is also used in other places like `waypoint_navigation_handler` and `nav_plan_init` where it may fail silently and return coordinates (0,0,0). I opened #283